File: p380.cpp
#include <bits/stdc++.h>

using namespace std;

class RandomizedSet {
private:
    unordered_set<int> data;
public:
    /** Initialize your data structure here. */
    RandomizedSet() {
    }
    
    /** Inserts a value to the set. Returns true if the set did not already contain the specified element. */
    bool insert(int val) {
        if(data.find(val) != data.end())
            return false;
        data.insert(val);
        return true;
    }
    
    /** Removes a value from the set. Returns true if the set contained the specified element. */
    bool remove(int val) {
        auto iter = data.find(val);
        if(iter != data.end()) {
            data.erase(iter);
            return true;
        } else {
            return false;
        }
    }
    
    /** Get a random element from the set. */
    int getRandom() {
        auto random_iter = next(begin(data), rand()%data.size());
        return *random_iter;
    }
};

File: p380.py
from random import randrange

class RandomizedSet:

    def __init__(self):
        self.arr = []
        self.map = {}

    def insert(self, val: int) -> bool:
        if val not in self.map:
            self.map[val] = len(self.arr)
            self.arr.append(val)
            return True
        return False

    def remove(self, val: int) -> bool:
        if val not in self.map:
            return False
        idx = self.map[val]
        self.arr[len(self.arr)-1], self.arr[idx] = self.arr[idx], self.arr[len(self.arr)-1]
        self.map[self.arr[idx]] = idx
        del self.map[val]
        del self.arr[-1]
        return True

    def getRandom(self) -> int:
        return self.arr[randrange(0, len(self.arr))]
